domain/state_test.go

```diff
@@ -28,3 +31,68 @@ func (s *stateSuite) TestStateBaseGetDBNilFactory(c *gc.C) {
 	_, err := base.DB()
 	c.Assert(err, gc.ErrorMatches, `nil getDB`)
 }
+
+func (s *stateSuite) TestStateBasePrepare(c *gc.C) {
+	f := s.TxnRunnerFactory()
+	base := NewStateBase(f)
+	db, err := base.DB()
+	c.Assert(err, gc.IsNil)
+	c.Assert(db, gc.NotNil)
+
+	// Prepare new query.
+	stmt1, err := base.Prepare("SELECT name AS &M.* FROM sqlite_schema", sqlair.M{})
+	c.Assert(err, gc.IsNil)
+	// Validate prepared statement works as expected.
+	err = db.Txn(context.Background(), func(ctx context.Context, tx *sqlair.TX) error {
+		results := sqlair.M{}
+		err := tx.Query(ctx, stmt1).Get(results)
+		if err != nil {
+			return err
+		}
+		c.Assert(results["name"], gc.Equals, "schema")
+		return nil
+	})
+	c.Assert(err, gc.IsNil)
+
+	// Retrieve previous statement.
+	stmt2, err := base.Prepare("SELECT name AS &M.* FROM sqlite_schema", sqlair.M{})
+	c.Assert(err, gc.IsNil)
+	c.Assert(stmt1, gc.Equals, stmt2)
+}
+
+func (s *stateSuite) TestStateBasePrepareKeyClash(c *gc.C) {
+	f := s.TxnRunnerFactory()
+	base := NewStateBase(f)
+	db, err := base.DB()
+	c.Assert(err, gc.IsNil)
+	c.Assert(db, gc.NotNil)
+
+	// Prepare statement with TestType.
+	{
+		type TestType struct {
+			Name string `db:"name"`
```

Comment:
```suggestion
			TblName string `db:"tbl_name"`
```

You really would get a different statement if we did this.

Reply:
Ah yea, no reason not to do that in the test, changed

Reply:
So for this specific change, SQLair actually will execute your query before it gives validates the output params, this is because behind the scenes `Get` really just calls the sqlair functions `Query.Iter(); Query.Next(); Query.Get()`. I'm going to discuses with Joe if we should change this.